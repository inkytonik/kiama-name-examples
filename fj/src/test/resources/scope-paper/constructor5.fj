// Adapted from class/constructors.spt:binary constructor
// FIXED - Add fields
class A extends Object {
  Object f;
  Object g;
  A(Object f, Object g) {
    super();
    this.f = f;
    this.g = g;
  }
}
new A(new Object(), new Object())
