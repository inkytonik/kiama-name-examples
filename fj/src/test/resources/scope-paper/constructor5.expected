src/test/resources/scope-paper/constructor5.fj:4:10:info: f def: Object
  Object f;
         ^
src/test/resources/scope-paper/constructor5.fj:5:10:info: g def: Object
  Object g;
         ^
src/test/resources/scope-paper/constructor5.fj:6:12:info: f def: Object
  A(Object f, Object g) {
           ^
src/test/resources/scope-paper/constructor5.fj:6:22:info: g def: Object
  A(Object f, Object g) {
                     ^
src/test/resources/scope-paper/constructor5.fj:12:1:info: new A(new  ...: A
new A(new Object(), new Object())
^
src/test/resources/scope-paper/constructor5.fj:12:7:info: new Object ...: Object
new A(new Object(), new Object())
      ^
src/test/resources/scope-paper/constructor5.fj:12:21:info: new Object ...: Object
new A(new Object(), new Object())
                    ^
A
